frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/create-steps.ts

```diff
@@ -0,0 +1,114 @@
+import { WizardState } from '../reducer';
+import { BackingStorageType, RHCS, StepsId } from '../../../constants/create-storage-system';
+
+export const createSteps = (
+  t: TFunction,
+  state: WizardState,
+  hasStorageCluster: boolean,
+): WizardStep[] => {
+  const { backingStorage, currentStep } = state;
+  const { count } = currentStep;
+
+  const commonSteps = {
+    capacityAndNodes: {
+      id: StepsId.CapacityAndNodes,
+      name: t('ceph-storage-plugin~Capacity and nodes'),
+    },
+    securityAndNetwork: {
+      id: StepsId.SecurityAndNetwork,
+      name: t('ceph-storage-plugin~Security and network'),
+    },
+    reviewAndCreate: {
+      id: StepsId.ReviewAndCreate,
+      name: t('ceph-storage-plugin~Review and create'),
+    },
+  };
+
+  const rhcsExternalProviderSteps = [
+    {
+      id: StepsId.ConnectionDetails,
+      canJumpTo: count >= 2,
+      name: t('ceph-storage-plugin~Connection details'),
+    },
+    {
+      id: StepsId.ReviewAndCreate,
+      canJumpTo: count >= 3,
+      ...commonSteps.reviewAndCreate,
+    },
+  ];
+
+  const nonRhcsExternalProviderStep = {
+    canJumpTo: count >= 2,
+    id: StepsId.CreateStorageClass,
+    name: t('ceph-storage-plugin~Create storage class'),
+  };
+
+  switch (backingStorage.type) {
+    case BackingStorageType.EXISTING:
+      return [
+        {
+          canJumpTo: count >= 2,
+          ...commonSteps.capacityAndNodes,
+        },
+        {
+          canJumpTo: count >= 3,
+          ...commonSteps.securityAndNetwork,
+        },
+        {
+          canJumpTo: count >= 4,
+          ...commonSteps.reviewAndCreate,
+        },
+      ];
+    case BackingStorageType.LOCAL_DEVICES:
+      return [
+        {
+          id: StepsId.CreateLocalVolumeSet,
+          canJumpTo: count >= 2,
+          name: t('ceph-storage-plugin~Create local volume set'),
+        },
+        {
+          canJumpTo: count >= 3,
+          ...commonSteps.capacityAndNodes,
+        },
+        {
+          canJumpTo: count >= 4,
+          id: StepsId.SecurityAndNetwork,
+          ...commonSteps.securityAndNetwork,
+        },
+        {
+          canJumpTo: count >= 5,
+          id: StepsId.ReviewAndCreate,
+          ...commonSteps.reviewAndCreate,
+        },
+      ];
+    case BackingStorageType.EXTERNAL:
+      if (backingStorage.externalProvider === RHCS) {
+        return rhcsExternalProviderSteps;
+      }
+      if (!hasStorageCluster) {
+        return [
+          nonRhcsExternalProviderStep,
+          { canJumpTo: count >= 3, ...commonSteps.capacityAndNodes },
+          {
+            canJumpTo: count >= 4,
+            ...commonSteps.securityAndNetwork,
+          },
+          {
+            canJumpTo: count >= 5,
+            ...commonSteps.reviewAndCreate,
+          },
+        ];
+      }
+      return [
+        nonRhcsExternalProviderStep,
+        {
+          canJumpTo: count >= 3,
+          ...commonSteps.reviewAndCreate,
```

Comment:
small confusion, if storagecluster is already there, we are just showing the step for creating storageclass and then moving to review and create ? 

Is it just to create FlashSystem CR ?

Reply:
Shouldn't the same be applicable to other cases if storagecluster exists or not ?

Reply:
It is Storage system creation, if OCS Storage System is already there we do not want to create that one.
RHCS itself uses Storage Cluster - an OCS Storage System, hence its only for external vendors